.github/workflows/performance-benchmarking.yml

```
@@ -0,0 +1,596 @@
+
+jobs:
+  should-run-benchmarks:
+    name: Check if benchmarks should run
+    runs-on: ubuntu-latest
+    outputs:
+      run_benchmarks: ${{ steps.check.outputs.run_benchmarks }}
+      benchmark_type: ${{ steps.check.outputs.benchmark_type }}
+    steps:
+      - name: 🔍 Check benchmark conditions
+        id: check
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:3:33: Double quote to prevent globbing and word splitting

(shellcheck)